### PR TITLE
Add tests for 2D UNet blocks

### PR DESCRIPTION
**Pull request by hchings**
## What does this PR do?
Add test cases for 2D UNet block. Issue: https://github.com/huggingface/diffusers/issues/202

### CODE REVIEW

**Comment by hchings** [comment]
Hi @patrickvonplaten,
1. Currently, I added tests for `unet_2d_blocks.py`. Can you take a look? If this is directionally fine, I'll add tests for `unet_1d_blocks.py`. Thanks! 

2. The output tensor for SimpleCrossAttnDown/Up blocks and a few Attn-related Unet blocks on MPS v.s. on CPU are very different (Max diff can be 0.7~1.2+). I've seen some discussion on PyTorch saying CPU and MPS floating point math can yield to differences, but I think atol > 5e-3 seems too high. Any guidance on why?

File: tests/test_unet_blocks_common.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        include_temb=True,
+        include_res_hidden_states_tuple=False,
+        include_encoder_hidden_states=False,
+        include_skip_sample=False,
+    ):
+        batch_size = 4
+        num_channels = 32
+        sizes = (32, 32)
+
+        torch.manual_seed(0)
```
Comment (patrickvonplaten) [inline tests/test_unet_blocks_common.py:52]:
```suggestion
        generator = torch.manual_seed(0)
```

File: tests/test_unet_blocks_common.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        hidden_states = torch.randn((batch_size, num_channels) + sizes).to(torch_device)
```
Comment (patrickvonplaten) [inline tests/test_unet_blocks_common.py:53]:
```suggestion
        shape = (batch_size, num_channels) + sizes
        hidden_states = randn_tensor(shape, device=device)
```

`randn_tensors` can be imported via:

```py
from diffusers.utils import randn_tensor
```

File: tests/test_unet_blocks_common.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        dummy_input = {"hidden_states": hidden_states}
+
+        if include_temb:
+            temb_channels = 128
+            dummy_input["temb"] = torch.randn(batch_size, temb_channels).to(torch_device)
+
+        if include_res_hidden_states_tuple:
+            torch.manual_seed(1)
```
Comment (patrickvonplaten) [inline tests/test_unet_blocks_common.py:61]:
```suggestion
            generator = torch.manual_seed(1)
```
And then same changes as in: https://github.com/huggingface/diffusers/pull/1945/files#r1068450338

**Review by patrickvonplaten — COMMENTED** [review]
Hey @hchings,

Thanks a lot for the improvements! They look great :-) 

We've recently changed to a more robust way of creating random tensors via  a new `randn_tensor` method: https://github.com/huggingface/diffusers/pull/1906

Could you maybe adapt the way random tensors are created as mentioned here: https://github.com/huggingface/diffusers/pull/1945/files#r1068450338? 

This would be great - apart from this it looks super nice! 
I don't think you'll have to change any of the expected values when doing the change :-)

**Comment by hchings** [comment]
@patrickvonplaten  Thanks for the inputs! I've made the changes and skipped 2 tests on MPS as it produces inconsistent results.

**Review by patrickvonplaten — APPROVED** [review]
Thanks a lot!

**Pull request merged**